packages/react-ui-core/src/DatePicker/__tests__/Calendar-test.js

```diff
@@ -1,110 +1,224 @@
 import React from 'react'
 import { shallow } from 'enzyme'
-import { format, getDaysInMonth } from 'date-fns'
-import Calendar from '../Calendar'
+import format from 'date-fns/format'
+import getDaysInMonth from 'date-fns/getDaysInMonth'
+import ThemedCalendar from '../Calendar'
+
+const theme = {
+  'Calendar_Date-active': 'Calendar_Date-active',
+  'Calendar_Date-disabled': 'Calendar_Date-disabled',
+}
+
+const Calendar = ThemedCalendar.WrappedComponent
+const DATE_FORMAT = 'MM-dd-YYYY'
 
 describe('Calendar', () => {
-  it('initialises with todays date if startDate prop is not defined', () => {
-    const calendar = shallow(<Calendar />)
-    const dateFormat = 'DD-MM-YYYY'
-    const todaysDate = format(new Date(), dateFormat)
-    expect(format(calendar.state().date, dateFormat)).toEqual(todaysDate)
+  const setup = props => (
+    shallow(
+      <Calendar
+        theme={theme}
+        {...props}
+      />
+    )
+  )
+
+  describe('date state', () => {
+    it('uses default of today date if no value', () => {
+      const wrapper = setup()
+      const date = wrapper.state().date
+      expect(format(date, DATE_FORMAT)).toEqual(format(new Date(), DATE_FORMAT))
+    })
+
+    it('uses `value` prop when passed in', () => {
+      const value = new Date('10/10/2018')
+      const wrapper = setup({ value })
+      const date = wrapper.state().date
+      expect(format(date, DATE_FORMAT)).toEqual(format(new Date('10/10/2018'), DATE_FORMAT))
+    })
   })
 
-  it('date can be formatted by prop', () => {
-    const calendar = shallow(<Calendar dateFormat="YYYY-MM-DD" />)
-    expect(format(calendar.state().date, 'YYYY-MM-DD')).toEqual(format(new Date(), 'YYYY-MM-DD'))
+  describe('dateFormat', () => {
+    it('uses default `dateFormat` when none passed', () => {
+      const wrapper = setup()
+      const date = wrapper.state().date
+      const dateFormat = wrapper.instance().props.dateFormat
+      expect(format(date, dateFormat)).toEqual(format(new Date(), dateFormat))
```

Comment:
I'm not sure this test does what it says it does. Shouldn't it be

```javascript
expect(format(date)).toEqual(format(date, DATE_FORMAT))
```

??

Reply:
Also, why are we testing this? Are we testing something on our end, or are we retesting date-fn/format?

Reply:
@mpelzsherman, this is testing against the default prop if none passed.  The format for the default is _not_ the DATE_FORMAT at the top of the screen.  

@orionstein we're just making sure no one removes the default `format` prop or modifies it. It might be a little overkill... 🤷‍♂️

Reply:
It's confusing to me because you're saying "if none passed", but it looks like you ARE passing in a dateFormat. Or maybe you mean "if no **date** is passed"? It's ambiguous. In either case, wouldn't it be clearer to pass in `null` or `undefined` explicitly, rather than forcing the reader to confirm that the wrapper data is undefined? The same applies to the 'date state' tests above BTW. It's just a suggestion, but I think this would make the tests more readable.